zcash_primitives/src/zip32.rs

```diff
@@ -409,6 +411,35 @@ impl ExtendedSpendingKey {
     pub fn default_address(&self) -> (DiversifierIndex, PaymentAddress) {
         ExtendedFullViewingKey::from(self).default_address()
     }
+
+    pub fn derive_internal(&self) -> Self {
+        let i = {
+            let fvk = FullViewingKey::from_expanded_spending_key(&self.expsk);
+            Blake2bParams::new()
+                .hash_length(64)
+                .personal(crate::zip32::ZIP32_SAPLING_INT_PERSONALIZATION)
+                .hash(&fvk.to_bytes())
```

Comment:
This isn't correct; `fvk.to_bytes()` doesn't include the encoding of `dk`. This should be `EncodeExtFVKParts(ak, nk, ovk, dk)`